pulsar/src/main/java/org/openconnectors/PulsarSink.java

```diff
@@ -0,0 +1,72 @@
+package org.openconnectors;
+
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.openconnectors.config.Config;
+import org.openconnectors.connect.ConnectorContext;
+import org.openconnectors.connect.SinkConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+import static org.openconnectors.config.ConfigUtils.verifyExists;
+
+public class PulsarSink implements SinkConnector<byte[]> {
+    private static final Logger LOG = LoggerFactory.getLogger(PulsarSink.class);
+
+    PulsarClient client;
+    Producer producer;
+
+    @Override
+    public CompletableFuture<Void> publish(Collection<byte[]> messages) {
+        try {
+            for (byte[] message : messages) {
+                producer.send(message);
+            }
+        } catch (PulsarClientException e) {
+            LOG.error("Error receiving message from pulsar producer", e);
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    @Override
+    public void flush() throws Exception {
+
```

Comment:
producer has a flush

Reply:
flush is a bit tricky here. Should we just store the last future that we have and just wait for that to complete?

Reply:
I just checked the producer doesn't have a flush method